src/components/SuperDrawer/components/SuperDrawerHeader.tsx

```diff
@@ -0,0 +1,63 @@
+import { ReactNode } from "react";
+import { createPortal } from "react-dom";
+import { useBeamContext } from "src/components/BeamContext";
+import { ButtonGroup } from "src/components/ButtonGroup";
+import { OpenInDrawerOpts } from "src/components/SuperDrawer/useSuperDrawer";
+import { Css } from "src/Css";
+import { useTestIds } from "src/utils";
+
+interface SuperDrawerHeaderProps {
+  children: ReactNode;
+  hideControls?: boolean;
+}
+
+interface SuperDrawerHeaderStructuredProps {
+  title: string;
+  left?: ReactNode;
+  right?: ReactNode;
+  hideControls?: boolean;
+}
+
+export function SuperDrawerHeader(props: SuperDrawerHeaderProps | SuperDrawerHeaderStructuredProps): JSX.Element {
```

Comment:
Thought: Pretty neat how React props & TS type unions/type-checking play nicely together like this.